server/src/main/java/org/cloudfoundry/identity/uaa/provider/IdpAliasFailedException.java

```diff
@@ -15,24 +15,51 @@ public IdpAliasFailedException(
             @NonNull final Reason reason,
             @Nullable final Throwable cause
     ) {
-        super(cause, ERROR, buildMessage(identityProvider, reason), reason.responseCode.value());
+        this(
+                identityProvider.getId(),
+                identityProvider.getIdentityZoneId(),
+                identityProvider.getAliasId(),
+                identityProvider.getAliasZid(),
+                reason,
+                cause
+        );
+    }
+
+    public IdpAliasFailedException(
+            @Nullable final String idpId,
+            @Nullable final String idzId,
+            @Nullable final String aliasId,
+            @Nullable final String aliasZid,
+            @NonNull final Reason reason,
+            @Nullable final Throwable cause
+    ) {
+        super(cause, ERROR, buildMessage(idpId, idzId, aliasId, aliasZid, reason), reason.responseCode.value());
     }
 
-    private static String buildMessagePrefix(@NonNull final IdentityProvider<?> idp) {
+    private static String buildMessagePrefix(
```

Comment:
I'm not sure what code smell this particular change is addressing, but I think having 4 parameters vs. a single parameter class is also a code smell.

Reply:
The reason I changed this is the new method `createNewAliasIdp` in `IdentityProviderEndpoints`. With just the other constructor of `IdpAliasFailedException` being present, we would have to pass the original IdP as a parameter just for the exception case (e.g., https://github.com/cloudfoundry/uaa/pull/2637/commits/e5f579179e6d1c0486cd814ac6b817cbae848b21#diff-90da5e48423637b46ed07b703063b4fca4a0ee47cca8286548c91941b532fd07R566), even though all of the information for the exception message is also present in the alias IdP.